swift/apple/FirezoneKit/Sources/FirezoneKit/Stores/AuthStore.swift

```diff
@@ -230,9 +237,33 @@ public final class AuthStore: ObservableObject {
       Task {
         await self.signOut()
       }
+      #if os(macOS)
+        self.showSignedOutAlertmacOS()
+      #endif
     }
   }
 
+  #if os(macOS)
+    private func showSignedOutAlertmacOS() {
+      let alert = NSAlert()
+      alert.messageText = "Your Firezone session has ended"
+      alert.informativeText = "Please sign in again to reconnect"
+      alert.addButton(withTitle: "Sign In")
+      alert.addButton(withTitle: "Cancel")
+      NSApp.activate(ignoringOtherApps: true)
+      let response = alert.runModal()
+      if response == NSApplication.ModalResponse.alertFirstButtonReturn {
+        Task {
+          do {
+            try await self.signIn()
+          } catch {
+            self.logger.error("Error signing in: \(error)")
+          }
+        }
+      }
+    }
+  #endif
+
```

Comment:
Would it make more sense to extract this function and the iOS equivalent into a single class / helper module (or two)?

The UI logic here feels a bit out of place in this file.